lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,20 +108,51 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        if 'username' in self.data:
+            return self.cleaned_data['username']
+        else:
+            return None
+
     def _clean_requested_user(self, cleaned_data, course_key):
         """
         Validates and returns the requested_user, while checking permissions.
         """
         requesting_user = self.initial['requesting_user']
         requested_username = cleaned_data.get('username', None)
 
-        if not requested_username:
+        if requesting_user.is_anonymous:
+            return self._verify_anonymous_user(cleaned_data, course_key)
+
+        if requested_username is None:
             return self._verify_no_user(requesting_user, cleaned_data, course_key)
         elif requesting_user.username.lower() == requested_username.lower():
             return self._verify_requesting_user(requesting_user, course_key)
         else:
             return self._verify_other_user(requesting_user, requested_username, course_key)
 
+    @staticmethod
+    def _verify_anonymous_user(cleaned_data, course_key):
+        """
+        Verifies form for when the requesting user is anonymous.
+        """
+        if cleaned_data.get('all_block', None) is not None:
+            raise PermissionDenied(
+                u"Anonymous users do not have permission to access all blocks in '{course_key}'."
+                    .format(course_key=six.text_type(course_key))
+            )
+
+        if cleaned_data.get('username', None) is not '':
```

Comment:
@xitij2000, when I send a request as an anonymous user to http://localhost:18000/api/courses/v2/blocks/?course_id=course-v1%3AedX%2BDemoX%2BDemo_Course, this condition is true and the below error message is returned. However, I haven't passed either the `username` field or the `all_blocks` field and in that case, the API should return the error about requiring one of those two fields to retain backward compatibility.